Improve Tessie documentation to Gold tier quality scale (#42998)
home-assistant/home-assistant.github.io: source/_integrations/tessie.markdown

The Tessie {% term integration %} exposes various commands and sensors from the Tesla vehicles and energy products connected to your [Tessie](https://tessie.com/) subscription.

## What this integration allows

This integration provides comprehensive control and monitoring of your Tesla vehicles and energy products through Home Assistant:

> ### Vehicle control

Reviewer: ```suggestion
### Vehicle control

```

> **Vehicle control:**

Reviewer: ```suggestion
### Vehicle control
```
We use [bold to markup UI strings](https://developers.home-assistant.io/docs/documenting/general-style-guide), not to highlight text or replace a header. You could use a header here. Or if you think this is to much, just remove the bold markup.

> **Vehicle control:**

Reviewer: ```suggestion
Vehicle control:
```
If you think a H3 is too much here, you could just remove the bold markup.

- Monitor battery level, range, charging status, and location
- Control climate (heating, cooling, seat heaters, steering wheel heater)
- Lock/unlock doors and control windows, trunk, frunk, and charge port
- Start/stop charging and set charge limits
- Enable Sentry mode, valet mode, and defrost mode
- Flash lights, honk horn, and trigger HomeLink
- Track vehicle location and navigation destination
- Install software updates

> ### Energy product monitoring and control

Reviewer: ```suggestion
### Energy product monitoring and control

```

> **Energy product monitoring and control:**

Reviewer: ```suggestion
### Energy product monitoring and control
```

> **Energy product monitoring and control:**

Reviewer: ```suggestion
Energy product monitoring and control:
```

- Monitor battery power, grid power, solar power, and load consumption
- View energy reserves and state of charge
- Control backup reserve and operating modes
- Manage charging from grid and storm watch settings
- Monitor grid services and virtual power plant (VPP) participation

## Compatible devices

This integration supports all Tesla vehicle models and energy products:

> ### Vehicles

Reviewer: ```suggestion
### Vehicles

```

> **Vehicles:**

Reviewer: ```suggestion
### Vehicles
```

- Model 3
- Model Y
- Model S
- Model X
- Cybertruck

> ### Energy products

Reviewer: ```suggestion
### Energy products

```

> **Energy products:**

Reviewer: ```suggestion
### Energy products
```

- Powerwall 2/3
- Powerwall+
- Solar Inverters

## Prerequisites

You must have an active [Tessie](https://my.tessie.com/) subscription, generate a [Tessie Access Token](https://my.tessie.com/settings/api) and grant Tessie access to your Tesla vehicle by generating a [Tesla Virtual Key](https://www.tesla.com/_ak/tessie.com).

{% include integrations/config_flow.md %}

> ## Configuration

Reviewer: ```suggestion
```

```

### Notify when charging is complete

This automation sends a notification when your vehicle has finished charging:

{% raw %}
```yaml
automation:
  - alias: "Notify when Tesla charging complete"
    trigger:
      - platform: state
        entity_id: binary_sensor.my_tesla_battery_charging
        from: "on"
        to: "off"
    condition:
      - condition: numeric_state
        entity_id: sensor.my_tesla_battery_level
        above: 79
    action:
      - action: notify.mobile_app
        data:
          message: "Tesla charging is complete at {{ states('sensor.my_tesla_battery_level') }}%"
```
{% endraw %}

## Troubleshooting

### Actions return errors

If a vehicle action returns an error in Home Assistant, you should first try to perform the same action in the Tessie app. The app will guide you through the steps to fix common issues like command signing or scopes.

### Command signing

Most Tesla vehicles manufactured in 2021 or later require command signing for security. Command signing ensures that commands sent to your vehicle are cryptographically signed and verified.

To enable command signing:

1. Open the Tessie app on your mobile device

> 2. Navigate to your vehicle settings

Reviewer: ```suggestion
1. Open the Tessie app on your mobile device.
2. Navigate to your vehicle settings.
```

3. Follow the prompts to set up a [Tesla Virtual Key](https://www.tesla.com/_ak/tessie.com)

> 4. This installs Tessie's unique encryption fingerprint on your vehicle

Reviewer: ```suggestion
3. Follow the prompts to set up a [Tesla Virtual Key](https://www.tesla.com/_ak/tessie.com).
    - This installs Tessie's unique encryption fingerprint on your vehicle.
```

Once configured, all commands from Tessie (including those sent through Home Assistant) will be signed with Tessie's private key. The vehicle verifies each command before execution, preventing unauthorized access even if your API token is compromised.

Command signing is compatible with:
- Model 3 and Model Y (all years)
- Model S and Model X (2021 or newer)

### Missing scopes or permissions

Tessie requires specific Tesla account permissions to function properly. If certain features aren't working, you may need to verify your account has granted the necessary scopes:

- **Vehicle Information** - Required to retrieve vehicle data (battery level, climate status, etc.)
- **Vehicle Location** - Required to track vehicle location
- **Vehicle Commands** - Required to control the vehicle (lock/unlock, climate, etc.)
- **Vehicle Charging Management** - Required to control charging
- **Energy Product Information** - Required to retrieve energy product data

> - **Energy Product Commands** - Required to control energy products (Powerwall, Solar)

Reviewer: ```suggestion
- Vehicle information: Required to retrieve vehicle data (battery level, climate status, etc.)
- Vehicle location: Required to track vehicle location
- Vehicle commands: Required to control the vehicle (lock/unlock, climate, etc.)
- Vehicle charging management: Required to control charging
- Energy product information: Required to retrieve energy product data
- Energy product commands: Required to control energy products (Powerwall, Solar)
```
Are these UI strings? If they are, please ignore my comment. If they are not, they should not use bold.
We use [sentence-style capitalization](https://learn.microsoft.com/en-us/style-guide/capitalization).

        state: "off"
    action:
      - action: switch.turn_on
        target:
          entity_id: switch.my_tesla_charge
```

### Stop charging when battery reaches target

This automation stops charging when the desired charge level is reached:

```yaml
automation:
  - alias: "Stop Tesla charging at 80%"
    trigger:
      - platform: numeric_state
        entity_id: sensor.my_tesla_battery_level
        above: 79
    condition:
      - condition: state
        entity_id: binary_sensor.my_tesla_battery_charging
        state: "on"
    action:
      - action: switch.turn_off
        target:
          entity_id: switch.my_tesla_charge
```

### Open garage door when arriving home

This automation opens your garage door when your Tesla arrives home:

```yaml
automation:
  - alias: "Open garage when Tesla arrives"
    trigger:
      - platform: zone
        entity_id: device_tracker.my_tesla_location
        zone: zone.home
        event: enter
    action:
      - action: cover.open_cover
        target:
          entity_id: cover.garage_door
```

### Notify when charging is complete

This automation sends a notification when your vehicle has finished charging:

> 

Reviewer: ```suggestion

{% raw %}
```

```yaml
automation:
  - alias: "Notify when Tesla charging complete"
    trigger:
      - platform: state
        entity_id: binary_sensor.my_tesla_battery_charging
        from: "on"
        to: "off"
    condition:
      - condition: numeric_state
        entity_id: sensor.my_tesla_battery_level
        above: 79
    action:
      - action: notify.mobile_app
        data:
          message: "Tesla charging is complete at {{ states('sensor.my_tesla_battery_level') }}%"
```

> 

Reviewer: ```suggestion
{% endraw %}

```

Energy product data should update regularly without restrictions.

## Known limitations

- **Passenger temperature control**: The passenger set temperature is shown as a sensor but cannot be changed through Tessie or Home Assistant. Only the driver's temperature can be controlled.
- **Field update frequency**: Some vehicle data fields may not update frequently depending on your Tessie subscription tier and field configuration settings.
- **No reconfiguration**: The integration cannot be reconfigured through the UI. To change your API token or settings, you must remove and re-add the integration.
- **Software updates**: Vehicle software updates can only be installed from Home Assistant after they have finished downloading to the vehicle.

## Diagnostics

The Tessie integration supports diagnostic data collection to help troubleshoot issues. If you're experiencing problems with the integration, you can download diagnostic information to include when reporting issues.

To download diagnostics:

> {% include integrations/diagnostics_download.md %}

Reviewer: ```suggestion
```
You could point to the troubleshooting documentation instead: /docs/configuration/troubleshooting/#debug-logs-and-diagnostics
